### PR TITLE
apriltag: add v3.4.2, remove old versions

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

File: recipes/apriltag/all/conanfile.py [hunk 1]
```diff
@@ -58,16 +58,30 @@ def source(self):
...
         tc.generate()
         if is_msvc(self):
             deps = CMakeDeps(self)
             deps.generate()
 
-    def build(self):
+    def _patch_sources(self):
         apply_conandata_patches(self)
+        # Fix DLL installation
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
```
Comment (uilianries) [inline recipes/apriltag/all/conanfile.py:75]: Created a PR to the upstream fixing it: https://github.com/AprilRobotics/apriltag/pull/300

Comment (RubenRBS) [reply recipes/apriltag/all/conanfile.py:75]: Thanks a lot!!!

File: recipes/apriltag/all/conanfile.py [hunk 2]
```diff
@@ -58,16 +58,30 @@ def source(self):
...
+                        "ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}",
+                        "ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}\n"
+                        "RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}")
+        # Skip the building and installation of examples
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                        "# Examples", "return()")
```
[the remark this answers is not in the document]
Comment (RubenRBS) [reply recipes/apriltag/all/conanfile.py:81]: Yeah I would prefer to have this as a patch too

**Comment by valgur** [comment]
Don't close.

**Review by AbrilRBS — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

- Replacing patches for NOMINMAX by Conan features was a good move
- Using CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS is acceptable since the upstream is enforcing the same: https://github.com/AprilRobotics/apriltag/blob/v3.4.0/CMakeLists.txt#L35
- Confirmed that pthread4w is no longer needed: https://github.com/AprilRobotics/apriltag/blob/v3.2.0/CMakeLists.txt#L49
- Good catch with the debug prefix: https://github.com/AprilRobotics/apriltag/blob/v3.3.0/CMakeLists.txt#L54

**Pull request merged**